crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -121,8 +121,26 @@ impl<'src> Parser<'src> {
             TokenKind::With => Stmt::With(self.parse_with_statement(start)),
             TokenKind::At => self.parse_decorators(),
             TokenKind::Async => self.parse_async_statement(),
-            TokenKind::Match => Stmt::Match(self.parse_match_statement()),
-            _ => self.parse_single_simple_statement(),
+            token => {
+                if token == TokenKind::Match {
+                    // Match is considered a soft keyword, so we will treat it as an identifier if
+                    // it's followed by an unexpected token.
+
+                    match self.classify_match_token() {
```

Comment:
Is this effectively an optimization? E.g., could we _always_ use `try_parse_match_statement` and fall back when it returns `None`?

Reply:
👍 Makes sense!